gp_emulator/multivariate_gp.py

```diff
@@ -192,32 +192,42 @@ def compress ( self, X ):
         """Project full-rank vector into PC basis"""
         return X.dot ( self.basis_functions.T ).T
 
-    def predict ( self, y, do_deriv=True ):
+    def predict ( self, y, do_deriv=True, do_unc=False ):
         """Prediction of input vector
         
         The individual GPs predict the PC weights, and these are used to 
         reconstruct the value of the function at a point `y`. Additionally,
-        the derivative of the function is also calculated. This is returned
-        as a `( N_params, N_full )` vector (i.e., it needs to be reduced 
-        along axis 1)
+        the derivative and second derivative of the function is also calculated. 
+        This is returned as a `( N_params, N_full )` vector (i.e., it needs 
+        to be reduced along axis 1)
         
         Parameters:
         y: array
             The value of the prediction point
         do_deriv: bool
             Whether derivatives are required or not
+        do_unc: bool
+            Whether the second order derivatives are required or not.
         """
         fwd = np.zeros ( self.basis_functions[0].shape[0] )
         y = np.atleast_2d ( y ) # Just in case
         if do_deriv:
             deriv = np.zeros ( ( y.shape[1], self.basis_functions.shape[1] ) )
+        if do_unc:
+            hess = np.zeros ( ( y.shape[1], self.basis_functions.shape[1] ) )
         for i in xrange ( self.n_pcs ):
             pred_mu, pred_var, grad = self.emulators[i].predict ( y )
             fwd += pred_mu * self.basis_functions[i]
             if do_deriv:
                 deriv += np.matrix(grad).T * np.matrix(self.basis_functions[i])
-        if do_deriv:
+            if do_unc:
+                hess += np.matrix(pred_var).T * np.matrix(self.basis_functions[i])                
```

Comment:
This will not work, as what we need here is the Hessian method from the individual emulators, so no ``pred_var``, but ``self.emulators[i].hessian (y )``. Note that as the latter is an Npar * Npar array, rearranging things might be required